style/CONTRIBUTE.md

```diff
@@ -124,3 +124,27 @@ restart the web server.
 
 Please prettify all files prior to submission. Run `make code_format` to format all
 code files with js-prettifier.
+
+## Highway Shield Contributor's Guide
+
+Highway shields are a key feature of the OpenStreetMap Americana style. This guide describes some of the style principles that contributors of highway shield artwork should consider when submitting new shields. The required elements are as follows:
+
+1. Shields are in .svg format, saved in "Optimized SVG" format in Inkscape
+2. Shields are 20px on the smallest dimension (this will be rasterized to 20px or 40x depending on display resolution)
+3. Shields are license-compatible (public domain or CC0)
+
+In addition, the following style guidelines are recommended:
+
+1. Use a 1.25px stroke size for line elements.
```

Comment:
I’ve been wondering about this stroke size: I would’ve thought it would prevent pixel alignment, potentially making the image blurry on 1× displays. A hairline stroke would be great for the edge of a sign that’s meant just for contrast rather than as a border in its own right, but I’m not sure about other line elements.

Reply:
When the SVG size was 40px, I just eyeballed 2.5px as a nice looking size by eye, so I just divided by two to get 1.25px.  However, I'm not married to that line width.

Reply:
When we go through the SVG->raster pipeline, is there a way to ensure that those vertical/horizontal lines remain aligned to the pixel grid?  Are there things we can turn on in Inkscape to ensure that alignment?  Clearly, a 1px horizontal line that's straddling two pixel rows won't achieve a nice result.

Reply:
That is something the designer of the SVG has to do.  The screenshots above are from [Affinity Designer](https://affinity.serif.com/en-us/designer/) with gridlines turned on and set to 1px.  It looks like Inkscape has a similar grid feature but I'm less familiar with Inkscape.

Reply:
Fixed in c1e7753